### PR TITLE
Remove legacy sessions

### LINKED ISSUE #7231: Remove Sessions page entirely
We've decided to remove Sessions in their current form, so the Sessions page is no longer accessible from the sidebar in the navigation redesign. Before this, it was also replaced by Recordings with feature flag `6050-remove-sessions`, which I believe is rolled out to all Cloud users, _but_ no self-hosted instances. This feature flag additionally controls some other places outside of the sidebar where we linked to Sessions, so self-hosted users may still see some mentions of Sessions.

The flag has been added 7 weeks ago so I propose going forward fully with this and removing code of the Sessions page, along with all links to it and with the flag.

Related: https://github.com/PostHog/posthog/issues/6140
CC @paolodamico

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

Closes #7231. Removes all the legacy code related to the Sessions page.
- In addition, we disable all API endpoints related to the Sessions page that will no longer be used.

## How did you test this code?
- Tested manually that the recordings, person page, person modal all work as expected.
- Frontend tests pass
- Wating on Cypress tests & backend tests results.

### CODE REVIEW

File: ee/clickhouse/views/events.py
Comment (paolodamico) [inline ee/clickhouse/views/events.py:148]: Unsure about this change. Should we keep the API endpoints? If the answer is no, should we remove all the filtering, queries, .... ? Any thoughts @macobo @rcmarron @alexkim205

Comment (macobo) [reply ee/clickhouse/views/events.py:148]: If it's dead then it should be killed. However @rcmarron owns this.

**Review by macobo — APPROVED** [review]

File: ee/clickhouse/views/events.py
Comment (paolodamico) [reply ee/clickhouse/views/events.py:148]: Are you in a rush to get this merge or can we wait for Rick to be online?

Comment (paolodamico) [reply ee/clickhouse/views/events.py]: Alright, I've removed all the legacy code from the backend as well but will wait for @rcmarron's review before merging as I'm removing a lot of stuff

File: posthog/migrations/0184_delete_sessionsfilter.py
Comment (paolodamico) [inline posthog/migrations/0184_delete_sessionsfilter.py:13]: Not sure if we should back anything up / keep the model around. Would the filters have any future use?

**Review by alexkim205 — APPROVED** [review]
frontend lgtm

File: posthog/migrations/0184_delete_sessionsfilter.py
Comment (rcmarron) [reply posthog/migrations/0184_delete_sessionsfilter.py:13]: This is a good question. On cloud, this feature has very limited usage (excluding our team, 29 session filters have been made), so I feel pretty comfortable simply deleting it.

I guess the risk is that some self-hosted users make extensive use of saved session filters, and they want to downgrade after upgrading to get back to the old experience. I feel like this likelihood is pretty small - maybe we include a clear note in the release notes? @paolodamico

**Review by rcmarron — APPROVED** [review]
This looks good to me - assuming we're good with the decision to remove the session filters.

I also removed a few additional apis/queries that are no longer used. Excited to clear all of this out.

File: posthog/migrations/0184_delete_sessionsfilter.py
Comment (paolodamico) [reply posthog/migrations/0184_delete_sessionsfilter.py:13]: Sounds like a plan @rcmarron !

**Pull request merged**
